lib/sendgrid/mailer.ex

```diff
@@ -58,7 +57,9 @@ defmodule SendGrid.Mailer do
       reply_to: email.reply_to,
       send_at: email.send_at,
       template_id: email.template_id,
-      mail_settings: %{ sandbox_mode: %{ enable: @sandbox_mode } }
+      mail_settings: %{ sandbox_mode: %{ enable: sandbox_mode } }
     }
   end
+
+  defp sandbox_mode, do: Application.get_env(:sendgrid, :sandbox_enable) || false
```

Comment:
I don't know. In my opinion this is a runtime config, not deps compile.

If the user change the setting it will you need to make the compilation of the dependency instead of the application. This makes it impossible, for example, that it can enable/disable during the execution of tests. Another thing about this, sure the users will not realize that is necessary to recompile the dependency.

Many dependencies prefer to make use of the approach that I suggest for these reasons. Not to mention that, in the end, this will be part of the release of the main application.

ex, ecto repo configs: https://github.com/elixir-ecto/ecto/blob/47aa0d0162eb75d1120260ce20387730b342f793/lib/ecto/repo/supervisor.ex#L18